test/ext/declarative/test_basic.py

```diff
@@ -2131,6 +2131,28 @@ def data_hybrid(self):
 
         assert not hasattr(Foo, "data_hybrid")
 
+    def test_kw_support_in_declarative_meta_init(self):
+        # This will not fail if DeclarativeMeta __init__ supports **kw
+
+        class BaseWithInitSubclass(Base):
+            __abstract__ = True
+
+            @classmethod
+            def __init_subclass__(cls, random_keyword_used_here=False, **kw):
+                pass
```

Comment:
maybe we could set `random_keyword_used_here` to the class so that we can test that the correct value was propagated? like `cls._set_random_keyword_used_here = random_keyword_used_here` then test it like
```py
eq_(User._set_random_keyword_used_here, True)
eq_(AnotherUser._set_random_keyword_used_here, False)
```

Reply:
yeah I wasn't too sure how to check the functionality beyond "I don't want this syntax to break the ORM class".

My main concern with actually _using_ this `__init_subclass__` functionality in this test is that it only exists in python 3.6+.  As it stands, the two test cases are:
 * `__init_subclass__` exists in the python version: 
   * the class method introduced in the test overrides the `<base>.__init_subclass__` so it can participate as a hook, as intended.
   * If this function is omitted, then the `<base>.__init_subclass__` complains that it doesn't take the provided `random_keyword_used_here` kwarg.
   * The `random_keyword_used_here` kwarg is then passed to `DeclarativeMeta.__init__`, where `**kw` eats it.
 * `__init_subclass__` doesn't exist in the python version:
   * We've now defined a totally new dunder class method `__init_subclass__` that doesn't get hooked anywhere.
   * The `random_keyword_used_here` kwarg is passed directly to `DeclarativeMeta.__init__`, where `**kw` still eats it.

If I understand the CI pipeline correctly, `sqlalchemy` tests against 3.8, so we're squarely in the first test case.  If we don't move to the second test case, I can make this test a little richer.

This is my first `sqlalchemy` PR: do you require that your tests work against lower versions of python as well?

Reply:
In addition, I'm afraid that the test, as I've written it, will not be clear on failure what went wrong.  Should I catch the `TypeError` and re-raise with a better error message?

Reply:
oh right.
Regarding the python version, you can use `util.py36` to check that python 3.6 or later is used. So you can add the `eq_` test inside an if that checks it.
regarding the reraise I don't think it's needed. The test seems clear enough.
We test python 2.7, 3.5-3.8

Reply:
yeah that sounds like the best approach, thanks for the feedback!  I've protected the detailed test in a `util.py36` check and used `cls._set_random_keyword_used_here ` as you suggested.

Please let me know if there is anything else.